### PR TITLE
MNT Deprecation notice if  param has wrong type

### PR DESCRIPTION
**Pull request by sabina-talipova**
## Parent issue
- https://github.com/silverstripe/silverstripe-framework/issues/10831

### CODE REVIEW

File: src/ORM/Search/SearchContext.php [hunk 1]
```diff
@@ -140,6 +141,13 @@ protected function applyBaseTableFields()
      */
     public function getQuery($searchParams, $sort = false, $limit = false, $existingQuery = null)
     {
+        if (!in_array(gettype($limit), ['array', 'NULL', 'string'])) {
+            Deprecation::notice(
+                '5.1',
+                'The parametr $limit should be type of array|string|null. $limit = null by default.'
```
Comment (emteknetnz) [inline src/ORM/Search/SearchContext.php]:
```suggestion
                '5.1.0',
                '$limit should be type of array|string|null'
```

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: src/ORM/Search/SearchContext.php (hunk 1 above)
Comment (sabina-talipova) [reply src/ORM/Search/SearchContext.php]: Updated

File: src/ORM/Search/SearchContext.php [hunk 2]
```diff
@@ -140,6 +141,13 @@ protected function applyBaseTableFields()
      */
     public function getQuery($searchParams, $sort = false, $limit = false, $existingQuery = null)
     {
+        if (!in_array(gettype($limit), ['array', 'NULL', 'string'])) {
```
Comment (emteknetnz) [inline src/ORM/Search/SearchContext.php]: Just occurred to me that this deprecation notice will fire when not passing in an explicit value and relying on the default value for `$limit` which would be super annoying

Can you also add in a check for `count(func_get_args()) >= 3` (I think this is correct) to validate that `$limit` was actually passed in.  Seems that `func_get_args()` won't include default values which is rather handy in this scenario

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: src/ORM/Search/SearchContext.php (hunk 2 above)
Comment (sabina-talipova) [reply src/ORM/Search/SearchContext.php]: Updated

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**